File: crates/silicube/tests/fixtures/sources/staggered_prompt.cpp
#include <iostream>
#include <string>

int main() {
    std::string name;
    std::cout << "What is your name?" << std::endl;
    std::getline(std::cin, name);
    std::cout << "Hello, " << name << "!" << std::endl;

    int num;
    std::cout << "Enter a number:" << std::endl;
    std::cin >> num;
    std::cout << "Double: " << (num * 2) << std::endl;
    std::cout << "Triple: " << (num * 3) << std::endl;

    std::string word;
    std::cin.ignore();
    std::cout << "Enter a word:" << std::endl;
    std::getline(std::cin, word);
    std::cout << "You said: " << word << std::endl;
    std::cout << "Done!" << std::endl;

    return 0;
}

File: crates/silicube/tests/fixtures/sources/staggered_prompt.py
import sys

sys.stdout.write("What is your name?\n")
sys.stdout.flush()
name = sys.stdin.readline().strip()
sys.stdout.write("Hello, " + name + "!\n")
sys.stdout.flush()

sys.stdout.write("Enter a number:\n")
sys.stdout.flush()
num = int(sys.stdin.readline().strip())
sys.stdout.write("Double: " + str(num * 2) + "\n")
sys.stdout.write("Triple: " + str(num * 3) + "\n")
sys.stdout.flush()

sys.stdout.write("Enter a word:\n")
sys.stdout.flush()
word = sys.stdin.readline().strip()
sys.stdout.write("You said: " + word + "\n")
sys.stdout.write("Done!\n")
sys.stdout.flush()
